### PR TITLE
feat: Add social-media use-cases for Facebook, Twitter, Instagram

### PR DESCRIPTION
**Pull request by jnv**
## Description

This is straight from [social-media-demo](https://github.com/superfaceai/social-media-demo) repository.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by jnv** [comment]
@janhalama I have ported this over, but haven't added any tests, could you please take over?

**Comment by janhalama** [comment]
> @janhalama I have ported this over, but haven't added any tests, could you please take over?

I am going to take over this PR and add tests.

File: grid/social-media/publish-post/maps/facebook.test.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { RecordingDefinitions } from '@superfaceai/testing';
+import { env } from 'process';
+
+import { publishPostTest } from './publish-post';
+
+const ACCESS_TOKEN_REDACTED_VALUE = 'PARAMS_pageAccessToken';
```
Comment (jnv) [inline grid/social-media/publish-post/maps/facebook.test.ts:6]: It's nitpicking for sure but I'd consider mentioning that the value has been redacted so it's clear from the recodings, e.g.:

```suggestion
const ACCESS_TOKEN_REDACTED_VALUE = 'PAGE_ACCESS_TOKEN_VALUE_REDACTED';
```

File: grid/social-media/publish-post/maps/facebook.test.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      }
+    }
+  });
+};
+
+const beforeRecordingLoad = function (recordings: RecordingDefinitions) {
+  recordings.forEach(recording => {
+    recording.path.replace(
+      ACCESS_TOKEN_REDACTED_VALUE,
+      env.FACEBOOK_ACCESS_TOKEN || 'xxx'
```
Comment (jnv) [inline grid/social-media/publish-post/maps/facebook.test.ts:35]: Is it correct to replace the page access token with user access token? :thinking:

File: grid/social-media/publish-post/maps/facebook.test.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { RecordingDefinitions } from '@superfaceai/testing';
+import { env } from 'process';
+
+import { publishPostTest } from './publish-post';
+
+const ACCESS_TOKEN_REDACTED_VALUE = 'PARAMS_pageAccessToken';
+
+const beforeRecordingSave = function (recordings: RecordingDefinitions) {
```
Comment (jnv) [inline grid/social-media/publish-post/maps/facebook.test.ts:8]: Would be helpful to explain rationale for this recordings filter, e.g.:

```
// Remove page access tokens from internal requests and responses
```

**Review by jnv — COMMENTED** [review]
Great work on those tests, just a few nitpicks.

File: grid/social-media/publish-post/maps/facebook.test.ts (hunk 2 above)
Comment (janhalama) [reply grid/social-media/publish-post/maps/facebook.test.ts:35]: Not it is not. Thank you for the catch.

File: grid/social-media/publish-post/maps/facebook.test.ts (hunk 1 above)
Comment (janhalama) [reply grid/social-media/publish-post/maps/facebook.test.ts]: value changed

File: grid/social-media/publish-post/maps/facebook.test.ts (hunk 2 above)
Comment (janhalama) [reply grid/social-media/publish-post/maps/facebook.test.ts]: fixed

File: grid/social-media/publish-post/maps/facebook.test.ts (hunk 3 above)
Comment (janhalama) [reply grid/social-media/publish-post/maps/facebook.test.ts:8]: added

**Review by martinalbert — COMMENTED** [review]
Found two things, otherwise looks great 👍🏻

File: grid/social-media/publish-post/maps/facebook.test.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      recording.path = recording.path.replace(
+        pageAccessToken,
+        PAGE_ACCESS_TOKEN_REDACTED_VALUE
+      );
+    }
+    if (recording.response) {
+      const response = recording.response as { access_token?: string };
+      if (response.access_token) {
+        pageAccessToken = response.access_token;
+        recording.response = {
```
[the remark this answers is not in the document]
Comment (janhalama) [reply grid/social-media/publish-post/maps/facebook.test.ts:22]: Yes it is. First recorded request contains page access token in body. The second request uses page access token to authenticate and the token has to be replaced in url path. When is the hook run for the first request pageAccessToken is not defined and we do not want to try to replace in in request path.

File: grid/social-media/publishing-profiles/profile.supr [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+  example Successful {
+    input {}
+
+    result {
+      id = "110658944785122",
+      imageUrl = "https://scontent-prg1-1.xx.fbcdn.net/v/t39.30808-1/cp0/p50x50/263679317_110659048118445_8251344428401368545_n.png?_nc_cat=103&ccb=1-5&_nc_sid=dbb9e7&_nc_ohc=kN4tKgxAWdYAX_B0I6K&_nc_ht=scontent-prg1-1.xx&edm=AGaHXAAEAAAA&oh=00_AT8KpGTOc-H8gJrtOJwvNasfXrW8I27s0h1T5a8_nxUeyg&oe=61EDD39E",
+      name = "Sf-test",
+      username = "sftest7904",
+    }
```
[the remark this answers is not in the document]
Comment (janhalama) [reply grid/social-media/publishing-profiles/profile.supr:44]: fixed

**Review by martinalbert — APPROVED** [review]

**Pull request merged**